src/Components/Web/src/Head/BasePath.cs

```diff
@@ -0,0 +1,119 @@
+/// </summary>
+public sealed class BasePath : ComponentBase
+{
+    private string _href = "/";
+
+    [Inject]
+    private NavigationManager NavigationManager { get; set; } = default!;
+
+    [Inject]
+    private IServiceProvider Services { get; set; } = default!;
+
+    /// <summary>
+    /// Gets or sets an explicit base path to render. When provided, this value takes precedence
+    /// over values inferred from the current request or fallback parameters.
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// &lt;BasePath Href="/dashboard/" /&gt;
+    /// </code>
+    /// </example>
+    [Parameter]
+    public string? Href { get; set; }
+
+    /// <summary>
+    /// Gets or sets the fallback <c>href</c> value to use when the current request information isn't available.
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// &lt;BasePath FallbackHref="/dashboard/" /&gt;
+    /// </code>
+    /// </example>
+    [Parameter]
+    public string? FallbackHref { get; set; }
+
+    /// <inheritdoc />
+    protected override void OnParametersSet()
+    {
+        _href = ComputeHref();
+    }
+
+    /// <inheritdoc />
+    protected override void BuildRenderTree(RenderTreeBuilder builder)
+    {
+        builder.OpenElement(0, "base");
+        builder.AddAttribute(1, "href", _href);
+        builder.CloseElement();
+    }
+
+    private string ComputeHref()
+    {
+        if (!string.IsNullOrEmpty(Href))
+        {
+            return Normalize(Href);
+        }
+
+        if (TryGetPathBase(out var pathBase))
+        {
+            return Normalize(pathBase);
+        }
+
+        if (!string.IsNullOrEmpty(FallbackHref))
+        {
+            return Normalize(FallbackHref);
+        }
+
+        var baseUri = NavigationManager.BaseUri;
+        if (Uri.TryCreate(baseUri, UriKind.Absolute, out var absoluteUri))
+        {
+            return Normalize(absoluteUri.AbsolutePath);
+        }
+
+        return "/";
+    }
+
+    private bool TryGetPathBase([NotNullWhen(true)] out string? pathBase)
+    {
+        var accessor = Services.GetService<IHttpContextAccessor>();
+        var contextPathBase = accessor?.HttpContext?.Request.PathBase;
+        if (contextPathBase.HasValue)
+        {
+            pathBase = contextPathBase.Value;
+            return true;
+        }
+
+        pathBase = null;
+        return false;
+    }
+
+    private static string Normalize(string value)
+    {
+        if (string.IsNullOrEmpty(value))
+        {
+            return "/";
+        }
+
+        var normalizedValue = Uri.TryCreate(value, UriKind.Absolute, out var absoluteUri)
+            ? absoluteUri.AbsolutePath
+            : value;
+
+        if (!normalizedValue.StartsWith('/'))
+        {
+            normalizedValue = "/" + normalizedValue;
+        }
+
+        return normalizedValue.EndsWith('/') ? normalizedValue : normalizedValue + "/";
+    }
```

Comment:
Won't be needed since this comes from the value provided by `HttpContext` in the end.